lib/bolt/transport/winrm.rb

```diff
@@ -99,6 +99,9 @@ def run_script(target, script, arguments, _options = {})
 
       def run_task(target, task, arguments, _options = {})
         input_method = task.input_method
+        if powershell_file?(task.executable)
+          input_method = 'powershell' if input_method == 'both'
```

Comment:
This feels a little weird to me. I think it's valid to put `both` explicitly in the metadata. I feel like we should only be setting `input_method` to `powershell` if nothing was explicitly added in metadata.

Reply:
I sensed the weirdness with this solution. I did it this way because I did not want to mess with the vendored puppet code. 

I am trying to figure out what changing the default input_method in puppet would look like. Could it be as simple as setting the input_method https://github.com/puppetlabs/puppet/blob/5b9f49b6f75a8453e496e2567bdc89a3a3129eb4/lib/puppet/pops/pcore.rb#L74 at provided link to something like "default" and then dealing with setting (and possibly validating) the input_method in the various transport implementations in bolt? 

What kind of impact would changing that default value have outside of the Bolt transport  implementations? It seems like the current default value of "both" is somewhat misleading given that there are at least three valid options (`stdin`, `environment`, `powershell`, + others in the future? ).